src/backend-examples/common/scala/org/macrogl/examples/backend/common/BasicShape.scala

```diff
@@ -0,0 +1,163 @@
+
+    def init(): Unit = {
+      print("Basic Shape: init")
+
+      implicit val mgl = systemInit()
+
+      // Prepare Data
+
+      val vertexSource = """
+        attribute vec3 position;
+
+        uniform mat4 projection;
+        uniform mat4 view;
+  
+        void main(void) {
+          gl_Position = view * projection * vec4(position, 1.0);
+        }
+        """
+      
+      val fragmentSource = """
+        #ifdef GL_ES
+        precision mediump float;
+        #endif
+ 
+        uniform vec3 color;
+  
+        void main(void) {
+          gl_FragColor = vec4(color, 1.0);
+        }
+        """
+
+      val vertexBufferData = Macrogl.createFloatData(12 * 3)
+      vertexBufferData.put(0.2f).put(0).put(-2)
+      vertexBufferData.put(0).put(0.2f).put(-2)
+      vertexBufferData.put(0).put(0).put(-2)
+      vertexBufferData.put(0).put(-0.2f).put(-2)
+      vertexBufferData.put(0.2f).put(0).put(-2)
+      vertexBufferData.put(0).put(0).put(-2)
+      vertexBufferData.put(0).put(0).put(-2)
+      vertexBufferData.put(-0.2f).put(0).put(-2)
+      vertexBufferData.put(0).put(-0.2f).put(-2)
+      vertexBufferData.put(0).put(0.2f).put(-2)
+      vertexBufferData.put(-0.2f).put(0).put(-2)
+      vertexBufferData.put(0).put(0).put(-2)      
+      vertexBufferData.rewind
+
+      val indicesBufferData = Macrogl.createShortData(3 * 1)
+      indicesBufferData.put(0.toShort).put(1.toShort).put(2.toShort)
+      indicesBufferData.rewind
+
+      var color = 0.0
+      val shapeColor = new Vector3f(0, 0.5f, 0.5f)
+      
+      val projectionTranform = Matrix.perspectiveProjection(50, width.toDouble / height, 0.1, 100.0)
+      
+      // General OpenGL
+      mgl.viewport(0, 0, width, height)
+      mgl.clearColor(1, 0, 0, 1) // red background
+
+      // Setup
+      val pp = new Program("BasicShape")(
+        Program.Shader.Vertex(vertexSource),
+        Program.Shader.Fragment(fragmentSource))
+      pp.acquire()
+
+      val vertexBuffer = new AttributeBuffer(Macrogl.STATIC_DRAW,
+        vertexBufferData.remaining() / 3, 3, Array((0, 3)))
+      vertexBuffer.acquire()
+      vertexBuffer.send(0, vertexBufferData)
+      
+      val camera = new Matrix.Camera(-0.25f, 0, 0);
+      
+      var continueCondition: Boolean = true
+      var direction: Int = 0
+      
+      def continue(): Boolean = {
+        continueCondition
+      }
+
+      def render(fe: FrameEvent): Unit = {
+        mgl.clear(Macrogl.COLOR_BUFFER_BIT)
+
+        for {
+          _ <- using.program(pp)
+          b <- using.vertexbuffer(vertexBuffer)
+        } {
+          pp.uniform.color = shapeColor
+          pp.uniform.projection = projectionTranform
+          pp.uniform.view = camera.transform
+          b.render(Macrogl.TRIANGLES)        
+        }
+
+        color += 0.01
+        shapeColor(2) = math.abs(color % 2.0 - 1.0).toFloat
+//        println(camera.position(0) , " " , camera.position(1) , " " , camera.position(2))
+        if(direction == 0){
```

Comment:
Please put one whitespace character before and after the parenthesis in an `if` statement.